module/userProjects/helper.js

```diff
@@ -2280,6 +2325,369 @@ module.exports = class UserProjectsHelper {
+     * certificate callback
+     * @method
+     * @name certificateCallback 
+     * @param {String} transactionId - transactionId for create certificate.
+     * @param {String} osid - osid for created certificate.
+     * @returns {JSON} certificate data updation details.
+    */
+
+    static certificateCallback(transactionId, osid) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                if ( transactionId == "" || osid == "" ) {
+                    throw {
+                        status: HTTP_STATUS_CODE["bad_request"].status,
+                        message: CONSTANTS.apiResponses.TRANSACTION_ID_AND_OSID_REQUIRED
+                    }
+                }
+                let updateObject = {
+                    "$set" : {}
+                };
+
+                // update osid and eligibility based on transactionId
+                updateObject["$set"]["certificate.osid"] = osid;
+                updateObject["$set"]["certificate.message"] = CONSTANTS.common.PROJECT_CERTIFICATE_GENERATED_SUCCESSFULLY;
+                updateObject["$set"]["certificate.issuedOn"] = new Date();
+
+                let projectDetails = await projectQueries.findOneAndUpdate(
+                    {
+                        "certificate.transactionId" : transactionId
+                    },
+                    updateObject,
+                    {
+                        new: true
+                    }
+                );
+
+                if ( projectDetails == null || !Object.keys(projectDetails).length > 0 ) {
+                    throw {
+                        status: HTTP_STATUS_CODE["bad_request"].status,
+                        message: CONSTANTS.apiResponses.PROJECT_NOT_FOUND
+                    }
+                }
+                 
+                return resolve({ 
+                    success: true,
+                    message: CONSTANTS.apiResponses.PROJECT_CERTIFICATE_GENERATED,
+                    data : {
+                        _id : ObjectId(projectDetails._id)
+                    }
+
+                });
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: {}
+                });
+            }
+        })
+    }
+
+    /**
+     * List user project details with certificate
+     * @method
+     * @name certificates 
+     * @param {String} userId - userId.
+     * @returns {JSON} certificate data updation details.
+    */
+
+    static certificates(userId) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                let certificateCount = 0;
+                //  get project details of user which have certificate.
+                const userProject = await projectQueries.projectDocument({
+                    userId: userId,
+                    status: CONSTANTS.common.SUBMITTED_STATUS,
+                    certificate: {$exists:true}
```

Comment:
@VISHNUDAS-tunerlabs just re-check this condition if its as per the requirement.